UPBGE: BL_Action: Add checks to try to ensure we're playing the right action

youle31 opened the pull request:
it is an attempt to fix https://github.com/UPBGE/upbge/issues/1445

There are 2 things:

- We compare directly action pointers instead of action names when we can.
- When it is not possible, for currently playable action "data types", we ensure that m_action (actuator's action) FCurve rna path contains the "data name" (constraint name, modifier name...) (if not the action is not played). The code is derived from BKE_modifier_uses_time and adapted.

BL_Action code can potentially be improved to support more action types but needs more investigations and can be done later.

lordloki commented:
For me it is not working after last commit. If I revert the last commit it works. The code seems OK. I checked the idprop action branch and it worked well too. Great!!

youle31 commented:
Ah, ok, lol, I'll check and test again, and commit without last commit. Thanks very much for review and tests!

youle31 commented:
Ok, Thanks very much for the checks you did. Indeed, the last commit seems to cause an issue. I don't understand why, but it is working as you said. I tried to improve the names checking, left debug infos commented just in case, reverted previous commit. I merge like that and if you see improvements or find a mistake in the code I added, we can fix later. Thanks very much for your eagle eye review!
